Handling of schemas whose fields using reserved keywords for Python variables

**Describe the bug**
I have avro schema which contains field name like yield and fake. The generated model contains yield and fake as a field. having fake field in the model create issue when using fake method and yield causes issue because its a reserved keyword in python.

fake field resulted in

`AssertionError: Invalid default type <class 'method'> for field "fake". Default should be <class 'str'>`

**To Reproduce**

```
import json

from dataclasses_avroschema import ModelGenerator, ModelType

model_generator = ModelGenerator()
schema = """
	{
		"type": "record",
		"name": "Test",
		"namespace": "com.foo",
		"fields": [{
			"name": "yield",
			"type": "double"
		}, {
			"name": "fake",
			"type": {
				"type": "string"
			}
		}]
	}
"""
new_schema = json.loads(schema)
result = model_generator.render(schema=new_schema, model_type=ModelType.AVRODANTIC.value, include_original_schema=True)
print(result)

==========
class Test(AvroBaseModel):
    yield: float
    fake: str

    
    class Meta:
        namespace = "com.foo"
        original_schema = """{"type": "record", "name": "Test", "namespace": "com.foo", "fields": [{"name": "yield", "type": "double"}, {"name": "fake", "type": {"type": "string"}}]}"""
```

**Expected behavior**
It should generate valid model perhaps use alias field

--- a/dataclasses_avroschema/model_generator/lang/python/base.py
+++ b/dataclasses_avroschema/model_generator/lang/python/base.py
@@ -1,6 +1,7 @@
 import copy
 import dataclasses
 import json
+import keyword
 import typing
 from abc import abstractmethod
 from dataclasses import dataclass, field
@@ -527,18 +528,28 @@ class BaseGenerator:
     @staticmethod
     def generate_field_name(field: JsonDict) -> str:
         field_name = field.get("name", "")
+        if not field_name:
+            return field_name
 
-        if field_name and not field_name.isidentifier():
+        if keyword.iskeyword(field_name):
+            valid_identifier = f"{field_name}_"
+            BaseGenerator.maybe_add_alias(field, field_name)
+            return valid_identifier
+
+        if not field_name.isidentifier():
             valid_identifier = casefy.snakecase(field_name)
             if valid_identifier != field_name:
-                aliases = field.get("aliases", [])
-
-                if valid_identifier not in aliases:
-                    aliases.append(field_name)
-                    field["aliases"] = aliases
-
+                BaseGenerator.maybe_add_alias(field, field_name)
             return valid_identifier
         return field_name
+
+    @staticmethod
+    def maybe_add_alias(field: JsonDict, alias: str) -> None:
+        aliases = field.get("aliases", [])
+
+        if alias not in aliases:
+            aliases.append(alias)
+            field["aliases"] = aliases
 
     @staticmethod
     def is_logical_type(*, field: JsonDict) -> bool:

--- a/tests/model_generator/conftest.py
+++ b/tests/model_generator/conftest.py
@@ -63,6 +63,20 @@ def schema_with_invalid_python_identifiers() -> Dict:
 
 
 @pytest.fixture
+def schema_with_python_keywords() -> Dict:
+    return {
+        "type": "record",
+        "name": "Address",
+        "fields": [
+            {"name": "class", "type": "string"},
+            {"name": "yield", "type": "string", "aliases": ["yield"]},
+            {"name": "yield_class", "type": "long"},
+        ],
+        "doc": "An Address",
+    }
+
+
+@pytest.fixture
 def schema_primitive_types_as_defined_types() -> Dict:
     return {
         "type": "record",

--- a/tests/model_generator/test_model_pydantic_generator.py
+++ b/tests/model_generator/test_model_pydantic_generator.py
@@ -115,6 +115,28 @@ class Address(AvroBaseModel):
     assert result.strip() == expected_result.strip()
 
 
+def test_avro_pydantic_python_keywords(schema_with_python_keywords: types.JsonDict) -> None:
+    expected_result = """
+from dataclasses_avroschema.pydantic import AvroBaseModel
+import pydantic
+
+
+
+class Address(AvroBaseModel):
+    \"""
+    An Address
+    \"""
+    class_: str = pydantic.Field(metadata={'aliases': ['class']})
+    yield_: str = pydantic.Field(metadata={'aliases': ['yield']})
+    yield_class: int
+
+
+"""
+    model_generator = ModelGenerator()
+    result = model_generator.render(schema=schema_with_python_keywords, model_type=ModelType.AVRODANTIC.value)
+    assert result.strip() == expected_result.strip()
+
+
 def test_avro_pydantic_model_with_meta_fields(
     schema_one_to_self_relationship: types.JsonDict,
 ) -> None:

